pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +270,71 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+	partitionList, err := t.GetUniquePartitionList(txn)
+	if err != nil {
+		return resources, stats, errors.Wrapf(err, "failed to get partition list from table:%v", t.tableName)
+	}
+
+	tablePrefix := "/" + t.tableName + "/"
+	iterOpt := badger.DefaultIteratorOptions
+	iterOpt.Prefix = []byte(tablePrefix)
```

Comment:
This prefix will optimize what internal badger tables need to be scanned.  But you are using the table prefix which is very large.  If you make a new iterator for each partition + key prefix you can make this prefix much tighter potentially.  I don't know if it will make a real difference.  Maybe worth a quick experiment?